File: utils/src/main/java/com/tc/utils/utils/helpers/DeviceInfo.java
package com.tc.utils.utils.helpers;

import android.content.Context;
import android.os.Build;
import android.provider.Settings;

public class DeviceInfo {
    public String getAndroidVersion() {
        String release = Build.VERSION.RELEASE;
        int sdkVersion = Build.VERSION.SDK_INT;
        return "Android SDK: " + sdkVersion + " (" + release + ")";
    }

    public String getDeviceName() {
        String manufacture = Build.MANUFACTURER;
        String model = Build.MODEL;
        if (model.startsWith(manufacture)) {
            return model;
        } else {
            return manufacture + " " + model;
        }
    }

    public String getDeviceId(Context context) {
        return Settings.Secure.getString(context.getContentResolver(),
                Settings.Secure.ANDROID_ID);
    }

}

File: utils/src/main/java/com/tc/utils/utils/helpers/DeviceInfo.kt
package com.tc.utils.utils.helpers

import android.content.Context
import android.os.Build
import android.provider.Settings

class DeviceInfo {
    val androidVersion: String
        get() {
            val release = Build.VERSION.RELEASE
            val sdkVersion = Build.VERSION.SDK_INT
            return "Android SDK: $sdkVersion ($release)"
        }
    val deviceName: String
        get() {
            val manufacture = Build.MANUFACTURER
            val model = Build.MODEL
            return if (model.startsWith(manufacture)) {
                model
            } else {
                "$manufacture $model"
            }
        }

    fun getDeviceId(context: Context): String {
        return Settings.Secure.getString(
            context.contentResolver,
            Settings.Secure.ANDROID_ID
        )
    }
}